### ISSUE TITLE
Syscollector is retrieving java update version incorrectly

### ISSUE THREAD

**Issue by DanLarios07** [wazuh/wazuh#11582]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.2.5 | Syscollector| Any| Any | Windows Server/Windows 10 |

Hi team!

I have found an issue in which the syscollector is not retrieving the `update` version of java correctly, below I provide the evidence of it:

## Version retrieved by syscollector:
```
0|2021/12/27 19:47:22|win|Java 8 Update 281 (64-bit)|||0|Oracle Corporation|20211227|8.0.2810.9|x86_64||||C:\Program Files\Java\jre1.8.0_281\|1|a:oracle:jre:8.0.2810.9::::::x86_64:||f8ba0a050772ffde38a9f43b109314d1d056d7b9|b1006540c0f56d9244b3f51e1f4731a67e4a2876
```

## Version on CPE:
On NVD article in the Known Affected Software Configurations section is stated this version:
`cpe:2.3:a:oracle:jre:1.8.0:update_281:*:*:*:*:*:*`
- Reference to the CVE: [CVE-2021-2161](https://nvd.nist.gov/vuln/detail/CVE-2021-2161)

## Rationale
The versions of the CPE and the version that the syscollector retrieves do not match, the version on CPE is 281 and the version retrieved by syscollector is 2810.

Thank you in advance

**Comment by jnasselle** [comment]
Hi @DanLarios07 ,

CPE information is populated by Vulnerability detector module using Syscollector information, which is consistent with Windows Registry information. This means that this behavior should be solved in Vulnerability detector module.

Regards,
Nico

**Comment by jctello** [comment]
To provide further insight here: 

It appears that Oracle uses its "JRE Security Baseline (Full Version String)" when publishing the CPEs associated to their CVEs but uses the simple version number on the Windows registry key of the installed application. 

So additional logic would be needed to fix this version mismatch (or for Oracle to decide to provide consistent versions)

**Comment by davidjiglesias** [comment]
Should be fixed in #14153
